PaulasWeatherApp/PaulasWeatherApp/ViewModel/LocationViewModel.swift

```diff
@@ -0,0 +1,74 @@
+//
+//  LocationViewModel.swift
+//  PaulasWeatherApp
+//
+//  Created by Stef, Paula on 8/23/22.
+//
+
+import Foundation
+import CoreLocation
+
+class LocationViewModel: NSObject {
+    private let notificationCenter: NotificationCenter
+    private let locationManager = { () -> CLLocationManager in
+        let manager = CLLocationManager()
+        manager.desiredAccuracy = kCLLocationAccuracyNearestTenMeters
+        manager.requestWhenInUseAuthorization()
+        manager.distanceFilter = 100.0
+        return manager
+    }()
+    let countries = NSLocale.isoCountryCodes.map { (code:String) -> String in
+        let id = NSLocale.localeIdentifier(fromComponents: [NSLocale.Key.countryCode.rawValue: code])
+        return NSLocale(localeIdentifier: "en_US").displayName(forKey: NSLocale.Key.identifier, value: id) ?? "Country not found for code: \(code)"
+    }
+    
+    init(notificationCenter: NotificationCenter = .default) {
+        self.notificationCenter = notificationCenter
+        super.init()
+    }
+    
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+    
+    func filterBy(countryName: String) -> [String] {
+        if !countryName.isEmpty {
```

Comment:
No need to change now, but in general if you have something on both _if_ branches you should verify the most simple case so that it's easy readable, like here you could test  if countryName.isEmpty and have the statements reversed